runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/TransformUpgrader.java

```diff
@@ -204,11 +211,22 @@ RunnerApi.Pipeline updateTransformViaTransformService(
 
     ExpansionApi.ExpansionRequest.Builder requestBuilder =
         ExpansionApi.ExpansionRequest.newBuilder();
+
+    // Creating a clone here so that we can set properties without modifying the original
+    // PipelineOptions object.
+    PipelineOptions optionsClone =
+        PipelineOptionsTranslation.fromProto(PipelineOptionsTranslation.toProto(options));
+    // Setting the option 'updateCompatibilityVersion' to the current SDK version so that the
```

Comment:
Maybe only set it if it was not explicitly set by the user (e.g. to an even older version)?

Reply:
Done.